tests/test_netstats.py

```diff
@@ -209,3 +234,240 @@ def test_raw_mets():
+    node_ci, ci, resolution, num_comms = netstats.community_resolution_selection(G)
+
+    assert len(node_ci) == len(ci)
+    assert num_comms == sim_num_comms
+    assert resolution is not None
+
+
+#@pytest.mark.parametrize("metric", ['rich_club_coeff'])
+@pytest.mark.parametrize("metric", ['participation', 'diversity', 'local_efficiency',
+                                    'comm_centrality', 'rich_club_coeff'])
+def test_get_metrics(metric):
+    """
+    Test various wrappers for getting nx graph metrics
+    """
+    base_dir = str(Path(__file__).parent/"examples")
+    est_path = f"{base_dir}/miscellaneous/0021001_rsn-Default_nodetype-parc_est-sps_thrtype-DENS_thr-0.19.npy"
+
+    in_mat = np.load(est_path)
+    G = nx.from_numpy_array(in_mat)
+    ci = np.ones(in_mat.shape[0])
+    metric_list_names = []
+    net_met_val_list_final = []
+
+
+    if metric == 'participation':
+        metric_list_names, net_met_val_list_final = \
+            netstats.get_participation(in_mat, ci, metric_list_names, net_met_val_list_final)
+        assert len(metric_list_names) == len(netstats.participation_coef(in_mat, ci))+1
+        assert len(net_met_val_list_final) == len(netstats.participation_coef(in_mat, ci))+1
+    elif metric == 'diversity':
+        metric_list_names, net_met_val_list_final = \
+            netstats.get_diversity(in_mat, ci, metric_list_names, net_met_val_list_final)
+        assert len(metric_list_names) == np.shape(netstats.diversity_coef_sign(in_mat, ci))[1]+1
+        assert len(net_met_val_list_final) == np.shape(netstats.diversity_coef_sign(in_mat, ci))[1]+1
+    elif metric == 'local_efficiency':
+        metric_list_names, net_met_val_list_final = \
+            netstats.get_local_efficiency(G, metric_list_names, net_met_val_list_final)
+        assert len(metric_list_names) == len(netstats.local_efficiency(G))+1
+        assert len(net_met_val_list_final) == len(netstats.local_efficiency(G))+1
+    elif metric == 'comm_centrality':
+        metric_list_names, net_met_val_list_final = \
+            netstats.get_comm_centrality(G, metric_list_names, net_met_val_list_final)
+        assert len(metric_list_names) == len(nx.algorithms.communicability_betweenness_centrality(G))+1
+        assert len(net_met_val_list_final) == len(nx.algorithms.communicability_betweenness_centrality(G))+1
+    elif metric == 'rich_club_coeff':
+        metric_list_names, net_met_val_list_final = \
+            netstats.get_rich_club_coeff(G, metric_list_names, net_met_val_list_final)
+        assert len(metric_list_names) == len(nx.algorithms.rich_club_coefficient(G))+1
+        assert len(net_met_val_list_final) == len(nx.algorithms.rich_club_coefficient(G))+1
+
+
+@pytest.mark.parametrize("plot_switch", [True, False])
+@pytest.mark.parametrize("sql_out", [True, False])
+@pytest.mark.parametrize("nc_collect", [True, False])
+@pytest.mark.parametrize("create_summary", [True, False])
+@pytest.mark.parametrize("graph_num", [1, 2])
+
```

Comment:
Will want to add an xfail where net_mets_csv_list is an empty list, as well as a case where the file is in the wrong format or not a graph.

Reply:
An empty list will raise an error but not and incorrect file format. 

When two (correct) files are passed in, an operational error will be raised when converting pd.DataFrame to a sql data base (core.utils.build_sql_db). I'm about to push a fix, but I don't really know why/how it's working.